pkg/scheduler/algorithm/priorities/util/topologies.go

```diff
@@ -17,22 +17,56 @@ limitations under the License.
 package util
 
 import (
+	"sync"
+
 	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-// GetNamespacesFromPodAffinityTerm returns a set of names
-// according to the namespaces indicated in podAffinityTerm.
-// If namespaces is empty it considers the given pod's namespace.
+// NamespaceLister helps list Namespaces.
+type namespaceLister interface {
+	// List lists all Namespaces in the indexer.
+	List(selector labels.Selector) (ret []*v1.Namespace, err error)
+}
+
+var (
+	once     sync.Once
+	nsLister namespaceLister
+)
+
+// InitNamespaceLister initialize the namespace lister
+func InitNamespaceLister(namespaceListerLister namespaceLister) {
+	once.Do(func() {
+		nsLister = namespaceListerLister
+	})
+}
+
+// GetNamespacesFromPodAffinityTerm return the namespaces from the pod affinity item
 func GetNamespacesFromPodAffinityTerm(pod *v1.Pod, podAffinityTerm *v1.PodAffinityTerm) sets.String {
```

Comment:
exposing a package function that depends on a global client doesn't seem appropriate